src/qt/guiutil.cpp

```diff
@@ -597,20 +600,40 @@ bool SetStartOnSystemStartup(bool fAutoStart)
             TCHAR pszExePath[MAX_PATH];
             GetModuleFileName(NULL, pszExePath, sizeof(pszExePath));
 
-            TCHAR pszArgs[5] = TEXT("-min");
+            // Start client minimized
+            QString strArgs = "-min";
+            // Start testnet client, if we are currently running testnet
+            if (Params().NetworkIDString() == "test")
+                strArgs += " -testnet";
+
+#ifdef UNICODE
+            // Convert the QString to TCHAR*
+            TCHAR* args = new TCHAR[strArgs.length() + 1];
```

Comment:
We should use RAII here, could be done in various ways. Most obvious is boost::scoped_array, e.g.

    scoped_array<TCHAR> args (new TCHAR[strArgs.length() + 1]);

Another option would be going through std::wstring, e.g.

    std::wstring wstrArgs = strArgs.toStdWString();
    TCHAR *args = W2T(wstrArgs.c_str());

In this case the manual appending of a NULL byte can be avoided, and the code can be shorter.